SAMPLE_FORMAT.md

```diff
@@ -25,19 +57,128 @@ specifically mentioned in the format, samples should make best-effort attempts i
 * __Idiomatic__ - examples should make best attempts to remain idiomatic and encourage good 
   practices that are specific to a language or practice. 
 
-## Format Guidelines
+## Build Tools
+
+All new samples should build and run integration tests with [Maven](https://maven.apache.org/).
+[Gradle](https://gradle.org/) support is optional as we don't yet have regular testing.
+
+## Testing
+Snippets should have integration tests that should verify the snippet works and compiles correctly.
+Creating mocks for these tests are optional. These tests should capture output created by the
+snippet to verify that it works correctly. See the tests in the canonical for an example of how to
+do this correctly. 
+
+* Test Library: [JUnit4](https://junit.org/junit4/)
+* Test Runner: [Maven Failsafe plugin](https://maven.apache.org/surefire/maven-failsafe-plugin/)
+and [Maven Surefire plugin](https://maven.apache.org/surefire/maven-surefire-plugin/).
 
+As an example, the following test code shows how we test a region tag called `region_tag`:
+```java
+package com.google.example;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Store region_tag in test method name, camel-cased
+ */ 
+@RunWith(JUnit4.class)
+public class SomeClassTest {
+  @Test public void regionTag_shouldPass() throws Exception { }
+
+  @Test public void regionTag_shouldFail() throws Exception {
+    Assert.fail("should fail");
+  }
+}
+```
+You will note that the "_" in `region_tag` is removed, and "_" is used to separate regionTags from
```

Comment:
This should read:
- underscores (`_`) are used in test method names to separate blocks of `camelCase` text ~from test descriptions~
- these blocks denote _region tags_, which serve as unique IDs for snippets in a given repository
- `camelCase` blocks beginning with `should` or `does` are **not** region tags - instead, they simply describe the test being run